library/CHART-TEMPLATE/templates/deployment.yaml

```diff
@@ -49,6 +49,12 @@ spec:
               port: http
           resources:
             {{- toYaml .Values.resources | nindent 12 }}
+          env:
+            - name: MY_APP_NON_OPTIONAL_ENV_VAR
+          {{- with include "common.proxy" . }}
+            - name: MY_APP_PROXY_URL
```

Comment:
I was wondering how to solve this, and I like the solution 👏 

(maybe it is obvious, but I did not think to it)

Reply:
I was not sure about the approach and I liked this that is simple and effective!